lib/ansible/modules/cloud/vmware/vmware_guest_cross_vc_clone.py

```diff
@@ -0,0 +1,292 @@
+        wait_for_task(task)
+        if task.info.state == 'error':
+            result = {'changed': False, 'failed': True, 'msg': task.info.error.msg}
+        else:
+            vm_info = self.get_new_vm_info(vm_name)
+            result = {'changed': True, 'failed': False, 'vm_info': vm_info}
+        return result
+
+    def sanitize_params(self):
+        '''
+        this method is used to verify user provided parameters
+        '''
+        self.vm_obj = find_vm_by_name(content=self.content, vm_name=self.params['name'])
+        if self.vm_obj is None:
+            vm_id = self.vm_uuid or self.vm_name or self.moid
+            self.module.fail_json(msg="Failed to find the VM/template with %s" % vm_id)
+
+        # connect to destination VC
+        self.destination_content = connect_to_api(
+            self.module,
+            hostname=self.destination_vcenter,
+            username=self.destination_vcenter_username,
+            password=self.destination_vcenter_password)
+        self.destination_datastore = find_datastore_by_name(content=self.destination_content, datastore_name=self.params['destination_datastore'])
+        if self.destination_datastore is None:
+            self.module.fail_json(msg="Destination datastore not specified by the user.")
+
+        self.destination_host = find_hostsystem_by_name(content=self.destination_content, hostname=self.params['destination_host'])
+        if self.destination_host is None:
+            self.module.fail_json(msg="Destination host not specified by the user.")
+
+    def populate_specs(self):
+        # populate service locator
+        self.service_locator.instanceUuid = self.destination_content.about.instanceUuid
+        self.service_locator.url = "https://" + self.destination_vcenter + ":" + str(self.params['port']) + "/sdk"
+        creds = vim.ServiceLocatorNamePassword()
+        creds.username = self.destination_vcenter_username
+        creds.password = self.destination_vcenter_password
+        self.service_locator.credential = creds
+
+        # populate relocate spec
+        self.relocate_spec.datastore = self.destination_datastore
+        self.relocate_spec.pool = self.destination_host.parent.resourcePool
+        self.relocate_spec.service = self.service_locator
+        self.relocate_spec.host = self.destination_host
+
+        # populate config spec
+        self.config_spec.numCPUs = 1
+        self.config_spec.memoryMB = 1024
+
+        # populate clone spec
+        self.clone_spec.config = self.config_spec
+        self.clone_spec.powerOn = self.module.params['power_on']
+        self.clone_spec.location = self.relocate_spec
+
+
+def main():
+    """
+    Main method
+    """
+    argument_spec = vmware_argument_spec()
+    argument_spec.update(
+        name=dict(type='str'),
+        uuid=dict(type='str'),
+        moid=dict(type='str'),
+        use_instance_uuid=dict(type='bool', default=False),
+        destination_vm_name=dict(type='str'),
```

Comment:
The documentation says these parameters are required but the argument spec says other way.